contracts/pools/PoolCollection.sol

```diff
@@ -594,10 +603,18 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
             pool,
             data,
             data.liquidity,
-            data.averageRate.rate.fromFraction112(),
+            data.averageRates.rate.fromFraction112(),
             _networkSettings.minLiquidityForTrading()
         );
 
+        // if trading is enabled, then update the recent average rates
+        if (data.tradingEnabled) {
+            _updateAverageRates(
+                data,
+                Fraction({ n: data.liquidity.bntTradingLiquidity, d: data.liquidity.baseTokenTradingLiquidity })
```

Comment:
By now, we probably need a pure getter function to take `PoolLiquidity` as input and return the spot-rate as output (can be used in a few places, including the one where we comprise the inverse spot-rate).

Reply:
I'll open a task. Closing for now.